### ISSUE TITLE
request for sub accounts

### ISSUE THREAD

**Issue by Oromis200** [moneymanagerex/moneymanagerex#3904]
# Description of the feature

It would be nice if it was possible to create sub accounts from your bank account (like with N26).

# Use cases

With sub accounts it would be possible to organize more complex financial structures or to enter information from bank accounts like N26 (which already implemented this feature).

**Comment by whalley** [comment]
Am I correct in understanding that you are talking about Spaces? : https://n26.com/en-eu/spaces

If so, why not create an account for each N26 space sub-account? E.g. N26-Main, N26-Emergency Fund, etc. ? Is there something else other than partitioning the account into sub-accounts that you need or is supported by N26 spaces?

**Comment by tactilis** [comment]
The approach suggested by @whalley is the easiest, but raises this question:

@Oromis200, when you export transactions from N26 and import them into MMEX, how does N26 handle transactions in/out of Spaces (also known as Pots, or Savings Pots in other banks)?

Does N26 produce a separate CSV (or QIF) file for each Space? Or is there just one file, which contains all the transactions from the N26 'main' account _and_ the Spaces?

**Comment by Oromis200** [comment]
@whally Indeed you understood correctly. So far I've been handeling it the way you described but the more spaces I created the more complex it became... Additionally because I have two bank accounts with multiple funds each. (e.g. if I want to check if my MMEX balance and the real account balance are the same I have to sum up every sub account manually.) This feature would basically add more visual clarity. 

@tactilis I'm not a N26 customer actually. I just wanted to use the setup as example :)
I enter information into MMEX manually because of my sub account (space) system.

**Comment by whalley** [comment]
@Oromis200 The introduction of user defined account grouping would help with this https://github.com/moneymanagerex/moneymanagerex/issues/3894 I think we should prioritise this.

**Comment by vomikan** [comment]
We can change the way accounts are grouped on the main page and the navigation bar. Group them not by type, but by "Held at". Thus, the accounts will be displayed not by type, but by bank.

![image](https://user-images.githubusercontent.com/6836805/152422507-b932882d-1862-466f-8263-a64044356600.png)

https://github.com/moneymanagerex/moneymanagerex/issues/3894

**Comment by whalley** [comment]
@vomikan I would see this as a 'bodge', as what we really want to do is provide a user-defined hierarchy for the accounts and a mechanism for filtering and reporting against this. What happens if the 'Held At' spans multiple account types? Yes, we could make it work but do we really want to do this now?  

My vote is to make 1.5.13 the last V1 release and move to V2 allowing us more freedom in database change.

**Comment by vomikan** [comment]
I totally agree.
I'm assigning a milestone just to just sort out the issue in the near future. 
Alternatively, we can start 1.16.x where we will add fields. for example, for the split or budget tables  field notes.

**Comment by georgeef** [comment]
The category hierarchy is of course a nice feature, but it is a mess for implementation. To get the name of a category, the whole hierarchy needs to be accessed. I think a similar functionality can be implemented for accounts in a simpler way, without changes in DB schema.

## Proposal

Account names can simply use a special symbol like `:` as a separator. From user's perspective, this separator defines a hierarchy of accounts. For *most* functions related to accounts, the separator means nothing, it is just part of account's name. For *some* functions specializing to account hierarchy, the separator is treated as a special symbol.

## Example

The user creates a few accounts with the following names:
- `A:main`
- `A:loan`
- `A:savings`

For *most* purposes, these accounts are independent and they can have any type.

Functions that show transactions, reports, etc., for an account, can be extended to interpret `A:` as the set of accounts starting with this prefix. The hierarchy relationship is not explicit in DB, it is implied by the special symbol `:` in the name.

**Comment by whalley** [comment]
Sounds a good suggestion. A few observations

- `:` is currently allowed in account names, we need to decide how to handle this
- This does not allow any characteristics to be added to an account 'grouping', e.g. if in the example above `A` represents a financial institution then the user could not store details about this. Though we could perhaps live with that.

**Comment by georgeef** [comment]
Yes, `:` is (and should be) allowed as a normal character in a name. The user can give this symbol a special meaning with an option is settings. E.g., "Account separator" can be empty, which means no hierarchy in accounts, or a symbol like `:` (better from a pre-defined list of choices), which means treat `:` as hierarchy separator. I don't see any problem if the user decides to change the separator, what was previous a set of independent accounts becomes a set of related accounts, and the inverse.

**Comment by georgeef** [comment]
This feature could make a few account types obsolete. The meaning of components in account names is left to the user. E.g., the first component could be "Loan:", which is another way to group all loan accounts together. The interface would become more dynamic and more difficult to program, but once implemented, users would have more flexibility on how to group accounts.

**Comment by georgeef** [comment]
Regarding
> This does not allow any characteristics to be added to an account 'grouping', e.g. if in the example above A represents a financial institution then the user could not store details about this. Though we could perhaps live with that.

Referring to the example above, users can create an account `A`, which may or may not contain transactions, where they can store all details about the "parent" account. [In the example above, this was supposed to be the purpose of `A:main`.]

In functions which process the separator symbol, `A` and all accounts staring with `A:` are treated as a group.
